packages/dev/docs/pages/react-aria/ssr.mdx

```diff
@@ -48,6 +48,234 @@ function App() {
+      config.plugins.push(localesPlugin.webpack({locales: []}));
+      /*- end highlight -*/
+    }
+    return config;
+  }
+};
+```
+
+Finally, add a `LocalizedStringProvider` to your root layout component. This example uses a URL parameter to get the requested locale. See the [Next.js Internationalization guide](https://nextjs.org/docs/app/building-your-application/routing/internationalization) to learn how to set this up.
+
+```tsx
+// app/[lang]/layout.tsx
+/*- begin highlight -*/
+import {LocalizedStringProvider} from 'react-aria-components/i18n';
+/*- end highlight -*/
+
+export default function RootLayout(
+  {children, params: {lang}}:
+  {children: React.ReactNode, params: {lang: string}}
+) {
+  return (
+    <html lang={lang}>
+      <body>
+        {/*- begin highlight -*/}
+        <LocalizedStringProvider locale={lang} />
+        {/*- end highlight -*/}
+        {children}
+      </body>
+    </html>
+  );
+}
+```
+
+**Note**: If you are using React Aria hooks rather than components, this can be imported from `react-aria/i18n` instead.
+
+#### Next.js Pages Router
+
+First, install `@react-aria/optimize-locales-plugin` with your package manager. Then, add the following to your `next.config.js`:
+
+```ts
+// next.config.js
+const localesPlugin = require('@react-aria/optimize-locales-plugin');
+
+module.exports = {
+  // ...
+  i18n: {
+    // See Next.js i18n docs...
+  },
+  webpack(config, {isServer}) {
+    if (!isServer) {
+      // Don't include any locale strings in the client JS bundle.
+      /*- begin highlight -*/
+      config.plugins.push(localesPlugin.webpack({locales: []}));
+      /*- end highlight -*/
+    }
+    return config;
+  }
+};
+```
+
+Finally, add a `LocalizedStringProvider` to `pages/_document.tsx` before the `<NextScript />` element. Use `props.locale` to access the requested locale, which is provided by Next.js. See the [Next.js Internationalization guide](https://nextjs.org/docs/pages/building-your-application/routing/internationalization) to learn how to set this up.
+
+```tsx
+// pages/_document.tsx
+import {Html, Head, Main, NextScript, DocumentProps} from 'next/document'
+ /*- begin highlight -*/
+import {LocalizedStringProvider} from 'react-aria-components/i18n';
+/*- end highlight -*/
+
+export default function Document(props: DocumentProps) {
+  return (
+    <Html lang={props.locale}>
+      <Head />
+      <body>
+        <Main />
+        {/*- begin highlight -*/}
+        <LocalizedStringProvider locale={props.locale} />
+        {/*- end highlight -*/}
+        <NextScript />
+      </body>
+    </Html>
+  );
+}
+
+```
+
+**Note**: If you are using React Aria hooks rather than components, this can be imported from `react-aria/i18n` instead.
+
+#### Remix
+
+Remix is supported [when using Vite](https://remix.run/docs/en/main/future/vite) for builds. First, install `@react-aria/optimize-locales-plugin` with your package manager. Then, add the following to your `vite.config.ts`:
```

Comment:
Mind adding a note here that LocalizedStringProvider isn't used here? Wasn't sure if the first two steps at the beginning of the "optimizing bundle size" were implicitly included here or not